NARFO_BE/NARFO_BE/Controllers/ActivitiesController.cs

```diff
@@ -0,0 +1,119 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using NARFO_BE.Models;
+
+namespace NARFO_BE.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ActivitiesController : ControllerBase
+    {
+        private readonly narfoContext _context;
+
+        public ActivitiesController(narfoContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Activities
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Activity>>> GetActivity()
+        {
+            return await _context.Activity.ToListAsync();
+        }
+
+        // GET: api/Activities/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Activity>> GetActivity(int id)
+        {
+            var activity = await _context.Activity.FindAsync(id);
+
+            if (activity == null)
+            {
+                return NotFound();
+            }
+
+            return activity;
+        }
+
+        // PUT: api/Activities/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutActivity(int id, Activity activity)
+        {
+            if (id != activity.ActiviryId)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(activity).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ActivityExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/Activities
+        [HttpPost]
+        public async Task<ActionResult<Activity>> PostActivity(Activity activity)
+        {
+            _context.Activity.Add(activity);
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateException)
+            {
+                if (ActivityExists(activity.ActiviryId))
+                {
+                    return Conflict();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return CreatedAtAction("GetActivity", new { id = activity.ActiviryId }, activity);
+        }
+
+        // DELETE: api/Activities/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<Activity>> DeleteActivity(int id)
+        {
+            var activity = await _context.Activity.FindAsync(id);
+            if (activity == null)
+            {
+                return NotFound();
+            }
+
+            _context.Activity.Remove(activity);
+            await _context.SaveChangesAsync();
```

Comment:
Why do some async save changes have a try catch around them, while others do not?

Reply:
This is scuffolded code.